geode-lucene/src/test/java/org/apache/geode/cache/lucene/LuceneQueriesReindexDUnitTest.java

```diff
@@ -51,7 +57,91 @@ private void recreateIndex() {
     LuceneIndexFactoryImpl indexFactory =
         (LuceneIndexFactoryImpl) luceneService.createIndexFactory().addField("text");
     indexFactory.create(INDEX_NAME, REGION_NAME, true);
-  };
+  }
+
+  @Test
+  @Parameters(method = "getListOfRegionTestTypes")
+  public void luceneQueryExecutedWithReindexWhenAllBucketsAreLostShouldNotCauseAHang(
+      RegionTestableType regionTestType) throws Exception {
+    dataStore1.invoke(() -> createDataStore(regionTestType));
+    dataStore2.invoke(() -> createDataStore(regionTestType));
+    accessor.invoke(() -> createAccessor(regionTestType));
+
+
+    putDataInRegion(accessor);
+    dataStore1.invoke(() -> Awaitility.await().atMost(1, TimeUnit.MINUTES)
```

Comment:
can we reduce this to 30s? I mean...for 3 entries... it should not take that long.. maybe even 10s

Reply:
It really shouldn't matter if the test runs correctly.  It should short circuit once the condition is met.